src/pages/Swap/index.tsx

```diff
@@ -394,6 +397,57 @@ export function Swap({
     return { amountIn: fiatValueTradeInput.data, amountOut: fiatValueTradeOutput.data }
   }, [fiatValueTradeInput, fiatValueTradeOutput])
 
+  const requireProof = async (type: 'swap' | 'wrap', callback?: () => any) => {
+    authorizationReqId.current = getRandomVcRequestId()
+    setShowVCsModalConf({ type, callback })
+
+    const date = new Date()
+    date.setFullYear(date.getFullYear() - 21)
+    const dateStringBefore21YearsFromNow = `${date.getFullYear()}${(date.getMonth() + 1)
+      .toString()
+      .padStart(2, '0')}${date.getDate().toString().padStart(2, '0')}`
+
+    const hardcodedAgeVCProofRequest = {
+      id: getRandomVcRequestId(),
+      circuitId: 'credentialAtomicQuerySigV2',
+      optional: false,
+      query: {
+        allowedIssuers: ['*'],
+        type: 'KYCAgeCredential',
+        context: 'https://raw.githubusercontent.com/iden3/claim-schema-vocab/main/schemas/json-ld/kyc-v3.json-ld',
+        credentialSubject: {
+          birthday: {
+            $lt: dateStringBefore21YearsFromNow,
+          },
+        },
+      },
+    }
+
+    const polygonId = new PolygonIdProvider()
+    const verifier = await polygonId.getVerifier()
+
+    try {
+      const proof = await provider?.send('torii_credential', hardcodedAgeVCProofRequest as any)
+
+      // TODO: find out mechanism to bind 'authorizationReqId' with the same
```

Comment:
actually, it's not just about matching the request id but if we really want to do a proper verification, we should validate that the proof matches the whole request which I didn't find a way to do yet with the SDK. But at least the onchain verification seems to do this matching. I opened an issue in the SDK repo: https://github.com/0xPolygonID/js-sdk/issues/118

A possible dirty workaround, at least for the circuit we use, could be to build a proof with dummy credentials here for the same query, extract the query hash from it (it's one of the fields in the raw proof structure) and compare it against the supplied proof, but I hope the devs will get back to me with a cleaner solution